clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -138,59 +132,60 @@ data TransformationInfo
   | TryTerm
   -- ^ Show the name and input to every transformation that is applied, and
   -- the result of every transformation that is applied.
-  deriving (Eq, Generic, Hashable, Ord, Read, Show, NFData)
+  deriving (Eq, Generic, Hashable, Ord, Read, Show, NFData, Enum, Bounded)
 
 -- | Options related to debugging. See 'ClashOpts'
 data DebugOpts = DebugOpts
-  { dbg_invariants :: Bool
+  { _dbg_invariants :: Bool
   -- ^ Check that the results of applied transformations do not violate the
   -- invariants for rewriting (e.g. no accidental shadowing, or type changes).
   --
   -- Command line flag: -fclash-debug-invariants
-  , dbg_transformationInfo :: TransformationInfo
+  , _dbg_transformationInfo :: TransformationInfo
   -- ^ The information to show when debugging a transformation. See the
   -- 'TransformationInfo' type for different configurations.
   --
   -- Command line flag: -fclash-debug-info (None|FinalTerm|AppliedName|AppliedTerm|TryName|TryTerm)
-  , dbg_transformations :: Set String
+  , _dbg_transformations :: Set String
   -- ^ List the transformations that are being debugged. When the set is empty,
   -- all transformations are debugged.
   --
   -- Command line flag: -fclash-debug-transformations t1[,t2...]
-  , dbg_countTransformations :: Bool
+  , _dbg_countTransformations :: Bool
   -- ^ Count how many times transformations are applied and provide a summary
   -- at the end of normalization. This includes all transformations, not just
-  -- those in 'dbg_transformations'.
+  -- those in '_dbg_transformations'.
   --
   -- Command line flag: -fclash-debug-count-transformations
-  , dbg_transformationsFrom :: Maybe Word
+  , _dbg_transformationsFrom :: Maybe Word
   -- ^ Debug transformations applied after the nth transformation applied. This
-  -- includes all transformations, not just those in 'dbg_transformations'.
+  -- includes all transformations, not just those in '_dbg_transformations'.
   --
   -- Command line flag: -fclash-debug-transformations-from=N
-  , dbg_transformationsLimit :: Maybe Word
+  , _dbg_transformationsLimit :: Maybe Word
   -- ^ Debug up to the nth applied transformation. If this limit is exceeded
   -- then Clash will error. This includes all transformations, not just those
-  -- in 'dbg_transformations'.
+  -- in '_dbg_transformations'.
   --
   -- Command line flag: -fclash-debug-transformations-limit=N
-  , dbg_historyFile :: Maybe FilePath
+  , _dbg_historyFile :: Maybe FilePath
   -- ^ Save information about all applied transformations to a history file
   -- for use with @clash-term@.
   --
   -- Command line flag: -fclash-debug-history[=FILE]
   } deriving (Generic, NFData, Show, Eq)
+makeLenses ''DebugOpts
```

Comment:
I wonder if instead of changing the names to every field, it would be better to just call `makeLenses` with options, i.e. tell it to generate a lens `foo` from a field `dbg_foo`. If you do this here and for `ClashOpts` the diff will be much shorter

Reply:
Don't we add `dbg_` to prevent naming collisions?

But even if we don't, `lens` is already complicated enough as it is, I think bringing in custom settings for smaller diffs isn't a good idea. The underscore-to-no-underscore pattern is pretty well established.

Reply:
> Don't we add dbg_ to prevent naming collisions?

No

> The underscore-to-no-underscore pattern is pretty well established.

Sure, it is more consistent to always use the same name scheme for lenses. It is a shame the default naming scheme is so intrusive though :cry:

Reply:
It is. We'll have somewhat sensible records usable in 2030 though :-).